azure/azure_kv.go

```diff
@@ -94,6 +97,9 @@ func (az *azureSecrets) GetSecret(
 		return nil, err
 	}
 
+	if secretResp.Value == nil {
+		return nil, ErrInvalidSecretResp
```

Comment:
You can use `secrets.ErrEmptySecretData` declared in secrets package at the top level.